### PR TITLE
support for tags added

### PR DESCRIPTION
**Pull request by bogdan-copocean**
Tags support added for outputs (#23).

### CODE REVIEW

File: share/config.py [hunk 1]
```diff
@@ -81,6 +83,9 @@ def __init__(
             shared_logger.warning("no namespace set in config: using `default`")
             self.namespace = "default"
 
+        if not self.tags or len(self.tags) == 0:
```
Comment (aspacca) [inline share/config.py:86]: `tags` is not mandatory

File: share/config.py [hunk 2]
```diff
@@ -81,6 +83,9 @@ def __init__(
...
+            shared_logger.warning("no tags set in config")
```
Comment (aspacca) [inline share/config.py:87]: no need for a `warning`, `debug` is fine.
while we are at it we can log the `tags`, regardless they are empty or not

File: tests/share/test_config.py [hunk 3]
```diff
@@ -727,3 +843,4 @@ def test_parse_config(self) -> None:
             assert elasticsearch.api_key == "api_key"
             assert elasticsearch.dataset == "dataset"
             assert elasticsearch.namespace == "namespace"
+            assert elasticsearch.tags == ["2021", "{'key1': 'value1'}", "['elem1', 'elem2']"]
```
Comment (aspacca) [inline tests/share/test_config.py:846]: good test case. but actually we should reject it :)
`tags` must be `list[str]`

Comment (bogdan-copocean) [reply tests/share/test_config.py:846]: should we reject every tag? let's say the user inputs `"tag1", 2, "tag3"`. We're dropping all tags or only the `2`?

File: shippers/es.py [hunk 4]
```diff
@@ -94,7 +96,7 @@ def _enrich_event(self, event_payload: dict[str, Any]) -> None:
 
         event_payload["event"] = {"dataset": self._dataset, "original": event_payload["fields"]["message"]}
 
-        event_payload["tags"] = ["preserve_original_event", "forwarded", self._dataset.replace(".", "-")]
+        event_payload["tags"] = self._tags
```
[the remark this answers is not in the document]
Comment (bogdan-copocean) [reply shippers/es.py:99]: merging like this: `["preserve_original_event", "forwarded", self._dataset.replace(".", "-")] + self._tags` ?

File: share/config.py (hunk 2 above)
Comment (bogdan-copocean) [reply share/config.py:87]: removed the if statement and added: `shared_logger.debug("tags: ", extra={"tags": self.tags})` or `shared_logger.debug(f"tags: {self.tags}")`

File: tests/share/test_config.py (hunk 3 above)
Comment (aspacca) [reply tests/share/test_config.py:846]: I would reject everything: it's a validation error.

Btw: we parse the config in the handler and use `Output` for the shipper factory, we should add a bad path test on `tests.handlers.aws.test_handler.TestLambdaHandlerFailure.test_lambda_handler_failure` for tags failing validation.

we should have done for the secret manager as well, I will open a ticket for it :)

File: share/config.py [hunk 5]
```diff
@@ -192,6 +197,29 @@ def id(self, value: str) -> None:
...
+        Tags setter.
+        It receives a list of tags and performs type validation
+        """
+        if not isinstance(values, list):
+            raise ValueError("Tags must be of type list")
+
+        for value in values:
+            if not isinstance(value, str):
+                raise ValueError(f"Each tag must be of type str: {value}")
+
```
Comment (aspacca) [inline share/config.py:220]: what about?
```
    self._tags = [value for value in values if isinstance(value, str)]
    if len(self._tags) != len(values):
        raise ValueError(f"Each tag must be of type str, given: {values}")
```

this way we print the original tags list

**Comment by aspacca** [comment]
@bogdan-copocean can you add `tags` to [README-AWS.md](https://github.com/elastic/elastic-serverless-forwarder/blob/main/docs/README-AWS.md)?

thanks

**Comment by bogdan-copocean** [comment]
yes, I'll add two more tests in handler and I'll write the docs too

File: .gitignore [hunk 6]
```diff
@@ -31,3 +31,4 @@ tests/elastic-serverless-forwarder-junit.xml
 *.code-workspace
 .pytest_cache/
 .python-version
+htmlcov/
```
Comment (aspacca) [inline .gitignore:34]: 👍

**Review by aspacca — APPROVED** [review]

**Pull request merged**